### ISSUE TITLE
Load multiple prepared datasets

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/power_perceiver#6]
## Detailed Description
A large part of my hope for the ML research we're doing in 2022 is to train across multiple "types" of prepared dataset. For example:

* just satellite (e.g. over the ocean);
* satellite + PV (e.g. over UK, Italy and Malta);
* satellite + PV + GSP + NWP (e.g. over Britain);
* (And, if the model overfits, then maybe try training on a few other video prediction datasets like "moving MNIST" or synthetic image sequences of clouds moving)

## Possible Implementation
Maybe implement a thin adaptor which holds multiple `power_perceiver.NowcastingDataset` instances, and itself inherits from `torch.utils.data.Dataset`. This thin adaptor would sample randomly sample from the upstream `power_perceiver.NowcastingDataset` instances and stack the `Tensors`. So for example, if we're combining "just satellite" data and "satellite + PV + GSP + NWP" then, say, the first 16 examples in each batch would be "just satellite", and the first 16 examples for PV, GSP, and NWP would be zeros (and would be masked out before it goes into the Perceiver).

**Comment by peterdudfield** [comment]
Sounds really great, and good to get it planned out. 

I would love to get https://github.com/openclimatefix/nowcasting_dataset/pull/562 into the new datasets, but I unfortunately Ive run out of time before my holiday
